code/modules/religion/sects/shadow_sect.dm

```diff
@@ -60,6 +64,88 @@
 	icon_state = "shadow-obelisk"
 	anchored = FALSE
 	break_message = "<span class='warning'>The Obelisk crumbles before you!</span>"
+	var/heal_delay = 20
+	var/last_heal = 0
+	var/spread_delay = 45
+	var/last_spread = 0
+
+/obj/structure/destructible/religion/shadow_obelisk/Initialize(mapload)
+	var/datum/religion_sect/shadow_sect/sect = GLOB.religious_sect
+	. = ..()
+	if(!sect.faithful_used)
+		return
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/structure/destructible/religion/shadow_obelisk/LateInitialize()
+	var/datum/religion_sect/shadow_sect/sect = GLOB.religious_sect
+	. = ..()
+	if(!sect.faithful_used)
+		return
+	START_PROCESSING(SSobj, src)
+
+/obj/structure/destructible/religion/shadow_obelisk/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/structure/destructible/religion/shadow_obelisk/process(delta_time)
+	var/datum/religion_sect/shadow_sect/sect = GLOB.religious_sect
+	if(sect.faithful_used)
+		if(last_heal <= world.time)
+			last_heal = world.time + heal_delay
+			for(var/mob/living/L in range(5, src))
+				if(L.health == L.maxHealth)
+					continue
+				if(!isshadow(L) && !L.mind?.holy_role && !isfaithless(L))
+					continue
+				new /obj/effect/temp_visual/heal(get_turf(src), "#29005f")
+				if(isshadow(L) || L.mind?.holy_role || isfaithless(L))
+					L.adjustBruteLoss(-2*delta_time, 0)
+					L.adjustToxLoss(-2*delta_time, 0)
+					L.adjustOxyLoss(-2*delta_time, 0)
+					L.adjustFireLoss(-2*delta_time, 0)
+					L.adjustCloneLoss(-2*delta_time, 0)
+					L.updatehealth()
+					if(L.blood_volume < BLOOD_VOLUME_NORMAL)
+						L.blood_volume += 1.0
+				CHECK_TICK
+		if(last_spread <= world.time)
+			var/list/validturfs = list()
+			var/list/shadowturfs = list()
+			for(var/T in circleviewturfs(src, 5))
+				if(istype(T, /turf/open/floor/black))
+					shadowturfs |= T
+					continue
+				var/static/list/blacklisted_pylon_turfs = typecacheof(list(
+					/turf/closed,
+					/turf/open/floor/black,
+					/turf/open/space,
+					/turf/open/lava,
+					/turf/open/chasm,
+					/turf/open/openspace,
+					/turf/open/floor/plating/beach,
+					/turf/open/indestructible,
+					/turf/open/floor/prison))
+				if(is_type_in_typecache(T, blacklisted_pylon_turfs))
+					continue
+				else
+					validturfs |= T
+
+			last_spread = world.time + spread_delay
+
+			var/turf/T = safepick(validturfs)
+			if(T)
+				if(istype(T, /turf/open/floor/plating))
+					T.PlaceOnTop(/turf/open/floor/black, flags = CHANGETURF_INHERIT_AIR)
+				else
+					T.ChangeTurf(/turf/open/floor/black, flags = CHANGETURF_INHERIT_AIR)
```

Comment:
What does this if statement actually do?

Reply:
Pretty much places the turf on top of open plating if there's no tile, otherwise, replaces the tile - stops people from Crowbaring the floor and opening space if I'm remembering right